### PR TITLE
Actual Specialist info (BUG) takes happiness changes into account

### PR DESCRIPTION
**Pull request by MeuhV1**

### CODE REVIEW

File: CvGameCoreDLL/CvGameTextMgr.cpp [hunk 1]
```diff
@@ -6492,6 +6492,19 @@ void CvGameTextMgr::parseSpecialistHelpActual(CvWStringBuffer &szHelpString, Spe
 			}
 			bStarted = setResumableCommerceTimes100ChangeHelp(szHelpString, szStart, L": ", L"", aiCommerces, true, bStarted);
 
+			// Happiness
+			if (iHappinessChange != 0)
+			{
+				if (iHappinessChange > 0)
```
Comment (dguenms) [inline CvGameCoreDLL/CvGameTextMgr.cpp]: You can remove that `if` statement and collapse the two `setResumableValueChangeHelp` calls (see other examples that display happiness), something like `setResumableValueChangeHelp(szHelpString, szStart, L": ", L"", std::abs(iHappinessChange), iHappiness > 0 ? gDLL->getSymbolID(HAPPY_CHAR) : gDLL->getSymbolID(UNHAPPY_CHAR), false, true, bStarted)`

**Review by dguenms — CHANGES_REQUESTED** [review]
Great idea, I'll also look at it within the game before approving.

**Comment by MeuhV1** [comment]
New code, same change

File: CvGameCoreDLL/CvGameTextMgr.cpp [hunk 2]
```diff
@@ -6492,6 +6492,9 @@ void CvGameTextMgr::parseSpecialistHelpActual(CvWStringBuffer &szHelpString, Spe
 			}
 			bStarted = setResumableCommerceTimes100ChangeHelp(szHelpString, szStart, L": ", L"", aiCommerces, true, bStarted);
 
+			// Happiness
+			bStarted = setResumableValueChangeHelp(szHelpString, szStart, L": ", L"", std::abs(iHappinessChange), iHappinessChange > 0 ? gDLL->getSymbolID(HAPPY_CHAR) : gDLL->getSymbolID(UNHAPPY_CHAR), false, true, bStarted);
```
Comment (dguenms) [inline CvGameCoreDLL/CvGameTextMgr.cpp:6496]: You still need the `iHappiness != 0` check though.

**Review by dguenms — CHANGES_REQUESTED** [review]

File: CvGameCoreDLL/CvGameTextMgr.cpp (hunk 2 above)
Comment (MeuhV1) [reply CvGameCoreDLL/CvGameTextMgr.cpp:6496]: There already is such a check in the setResumableValueChangeHelp function itself, so I could leave it out. Tested it and it works.

Comment (dguenms) [reply CvGameCoreDLL/CvGameTextMgr.cpp:6496]: Alright!

**Pull request merged**